src/errorCalculation.js

```diff
@@ -1,22 +1,151 @@
+import nd from 'norm-dist';
+
+/**
+ * @private
+ * @param {number} x
+ * @return {number}
+ */
+const sq = (x) => x * x;
```

Comment:
FYI, this isn't [really any faster](https://jsperf.com/math-square) than `Math.pow(x, 2)`, which has been part of the ES standard since the early days. (You can also [use `x ** 2` if you don't care about IE](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Exponentiation).)

Reply:
I doubt that the code in the perf actually does anything. Since you're effectively throwing away the result right after the computation, the interpreter probably optimizes that part out and just iterates `i` (probably by more than `1`).

You're probably right about this function not adding very much to performance. If you don't like it, just remove it, but it seems quite readable to me.

Reply:
I guess what I'm trying to say is that if you're doing it for performance reasons then it'd probably [better to remove it](http://wiki.c2.com/?PrematureOptimization). If you think it is more readable I can see why that might be a good reason. However, I think it's hard to argue that a custom function that is identical to a built-in one is easier to read (because one has to look it up to see what it does rather than just recognize it), though in this case it matters little either way since the function is so simple.

You are probably right about jsPerf...sometimes it's too easy to fool oneself. Here's an [updated example](https://jsperf.com/math-square/7) adapted to reduce the likelihood of this kind of mistake. It still shows less than 1% difference between the two methods of interest.
![image](https://user-images.githubusercontent.com/1633459/47392762-2024c100-d6e3-11e8-88aa-d47a8a62cdaa.png)

Reply:
You're right, the performance differences are negligible. I would choose `x ** 2` then, as it seems to have the best readability. And when we move to TypeScript, it will be [compiled to `Math.pow`](http://www.typescriptlang.org/play/#src=declare%20let%20x%3A%20number%3B%0Ax%20%3D%20x%20**%202%3B) so we won't lose compatibility with old browsers.